### PR TITLE
example: dissolve shader

### PR DESCRIPTION
**Pull request by kyeett**
Noise image used:
noise

Result
dissolve

### CODE REVIEW

File: examples/shader/dissolve.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build ignore
+
+package main
+
+var Time float
+var Cursor vec2
+var ImageSize vec2
```
Comment (hajimehoshi) [inline examples/shader/dissolve.go:21]: This was reverted to `ScreenImage`. Sorry!

File: examples/shader/dissolve.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+func Fragment(position vec4, texCoord vec2, color vec4) vec4 {
+	// Triangle wave to go 0-->1-->0...
+	Limit := abs(2*fract(Time/3) - 1)
```
Comment (hajimehoshi) [inline examples/shader/dissolve.go:25]: Use small caps for local variables.

File: examples/shader/dissolve.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	Level := image3TextureAt(texCoord).x
+
+	// Add a white border
+	if Level < Limit && Level > Limit-0.1 {
+		Alpha := image0TextureAt(texCoord).w
+		return vec4(1) * Alpha
```
Comment (hajimehoshi) [inline examples/shader/dissolve.go:31]: `vec4(alpha)`?

File: examples/shader/dissolve.go (hunk 1 above)
Comment (kyeett) [reply examples/shader/dissolve.go:21]: since this is not used, this is only for consistency?

File: examples/shader/dissolve.go (hunk 3 above)
Comment (kyeett) [reply examples/shader/dissolve.go]: yeah. I tried to convey the meaning "draw white, if alpha is zero", but it is hard to understand from the code above anyway :-) 

I will use vec4(alpha)

File: examples/shader/dissolve.go (hunk 2 above)
Comment (kyeett) [reply examples/shader/dissolve.go]: done

File: examples/shader/dissolve.go (hunk 1 above)
Comment (hajimehoshi) [reply examples/shader/dissolve.go]: This is only for consistency.

**Comment by hajimehoshi** [comment]
BTW, would it be possible to use OpenSimplex noise instead of adding a new image?

https://en.wikipedia.org/wiki/OpenSimplex_noise
https://github.com/ojrac/opensimplex-go/blob/master/opensimplex_base.go#L17-L128

**Comment by kyeett** [comment]
It is possible, but perhaps not very convenient. In a real world case, I would use this approach of sending in a pre-rendered image, as it is faster (and straight forward)

If we don't want to send in an image we must
1. Send it in as a uniform (not sure what format) 
OR
2. Make the simplex function inside the shader. 

You suggest 2nd option?

**Comment by hajimehoshi** [comment]
I suggested 2. OK I understand that the shader would be more complex. Let's go with binding-an-image way.

BTW, what is the license of the noise.png? Could you modify examples/resources/images/license.md?

**Comment by kyeett** [comment]
I don't remember where I got it from! I'll look, or generate a new one with a new source.

**Comment by kyeett** [comment]
Created a new noise image using OpenSimplex. It is not exactly the same, but I like this look as well.

noise

dissolve2

**Pull request merged**